quotes.py

```diff
@@ -0,0 +1,233 @@
+
+        args = utils.get_args(message)
+        reply = await message.get_reply_message()
+
+        if not reply:
+            return await utils.answer(message, self.strings["no_reply"])
+
+        if not args:
+            return await utils.answer(message, self.strings["no_template"])
+
+        ProfilePhotoURL = UsernameColor = Username = AdminTitle = UserID = Channel = None
+
+        if not reply.fwd_from:
+            user = await self.client(telethon.tl.functions.channels.GetParticipantRequest(message.chat_id,
+                                                                                          reply.from_id))
+            user_full = await self.client(telethon.tl.functions.users.GetFullUserRequest(reply.from_id))
+            if not await self.client.get_profile_photos(reply.from_id):
+                ProfilePhotoURL = ""
+
+            AdminTitle = getattr(user.participant, "participant", "")
+            if AdminTitle is None:
+                AdminTitle = ""  # API requirement
+
+            Username = telethon.utils.get_display_name(user_full.user)
+            UserID = reply.from_id
+        else:
+            if reply.fwd_from.from_name:
+                ProfilePhotoURL = ""
+                Username = reply.fwd_from.from_name
+            elif reply.fwd_from.channel_id:
+                try:
+                    cid = reply.fwd_from.channel_id
+                    resolved = (await self.client(telethon.tl.functions.channels.GetFullChannelRequest(cid))).chats[0]
+
+                    if isinstance(resolved.photo, telethon.tl.types.ChatPhotoEmpty):  # get_profile_photos won't work
+                        ProfilePhotoURL = ""
+
+                    AdminTitle = "Channel"
+                    Username = telethon.utils.get_display_name(resolved)
+                    UserID = cid
+                    Channel = resolved
+                except telethon.errors.rpcerrorlist.ChannelPrivateError:
+                    # we don't know anything about this channel.
+                    ProfilePhotoURL = ""
+                    Username = "Hidden."
+                AdminTitle = "Channel"  # 100%
+            elif reply.fwd_from.from_id:
+                try:
+                    UserID = reply.fwd_from.from_id
+                    resolved = (await self.client(telethon.tl.functions.users.GetFullUserRequest(UserID))).user
+
+                    if not await self.client.get_profile_photos(UserID):
+                        ProfilePhotoURL = ""
+
+                    Username = telethon.utils.get_display_name(resolved)
+                    AdminTitle = ""
+                except ValueError:
+                    ProfilePhotoURL = ""
+                    Username = "A user."
+                    AdminTitle = "User"
+
+        if ProfilePhotoURL != "" or Channel:
+            with BytesIO() as pfp:
+                if Channel:
+                    await self.client.download_profile_photo(Channel, pfp)
```

Comment:
If you're using download_profile_photo, it's more efficient do `download_profile_photo(..., bytes)` and it returns a `bytes` object

Reply:
do it yourself if you want to, I'm not going to